domain/domain.go

```diff
@@ -1022,6 +1022,9 @@ func (do *Domain) Close() {
 	if intest.InTest {
 		infosync.MockGlobalServerInfoManagerEntry.Close()
 	}
+	if handle := do.statsHandle.Load(); handle != nil {
```

Comment:
Is this change necessary? If it is not necessary for this PR, I prefer to keep it unchanged to avoid any risk 🤦

Reply:
It is to avoid the ```goleak```. so that we do not add ```ristretto``` into goleak config in the every ```main_test```.